source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -523,6 +526,9 @@ Http::FilterHeadersStatus JsonTranscoderFilter::decodeHeaders(Http::RequestHeade
           absl::StrCat(RcDetails::get().GrpcTranscodeFailedEarly, "{BAD_REQUEST}"));
       return Http::FilterHeadersStatus::StopIteration;
     }
+    stats_->transcoder_request_buffer_bytes_.add(initial_request_data_.length() -
+                                                 buffer_size_before);
+    stats_->transcoder_request_buffer_bytes_.sub(stream_size_before - request_in_.bytesStored());
```

Comment:
can we modify the atomic gauge once rather than twice?

Reply:
Done.